crates/common/logged_command/src/logged_command.rs

```diff
@@ -131,30 +131,31 @@ fn send_signal_to_stop_child(child: Option<u32>, signal_type: CmdStatus) {
 /// A command which execution is logged.
 ///
 /// This struct wraps the main command with a nice representation of that command.
-/// This `command_line` field is only required because the
-/// [`Command::get_program()`](https://doc.rust-lang.org/std/process/struct.Command.html#method.get_program)
-/// and
-/// [`Command::get_args()`](https://doc.rust-lang.org/std/process/struct.Command.html#method.get_args)
-/// are nightly-only experimental APIs.
 pub struct LoggedCommand {
-    command_line: String,
     command: Command,
 }
 
 impl std::fmt::Display for LoggedCommand {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.command_line.fmt(f)
+        let command = self.command.as_std();
+
+        command.get_program().to_string_lossy().fmt(f)?;
+        for arg in command.get_args() {
+            // The arguments are displayed as debug, to be properly quoted and distinguished from each other.
+            write!(f, " {:?}", arg.to_string_lossy())?;
+        }
+        Ok(())
     }
 }
 
 impl LoggedCommand {
+    /// Creates a new `LoggedCommand`.
+    ///
+    /// In contrast to [`std::process::Command`], `program` can contain space-separated arguments,
+    /// which will be properly parsed, split, and passed into `.args()` call for the underlying
+    /// command.
```

Comment:
note: Since `LoggedCommand` mirrors stdlib's `Command` API a bit, wanted to underline that passing a `program` with spaces to this constructor is fine. Now that we have `impl From<StdCommand>`, it may be tricky to remember the behaviour.